script/hassfest/triggers.py

```diff
@@ -26,21 +26,99 @@ def exists(value: Any) -> Any:
+                        f"Allowed keys: {', '.join(sorted(allowed_keys)) if allowed_keys else 'none'}"
+                    )
+
+                # Check if the referenced field exists in trigger schema or target
+                if not isinstance(field_ref, str):
+                    raise vol.Invalid(
+                        f"Context value for '{context_key}' must be a string field reference"
+                    )
+
+                # Check if field exists in trigger schema fields or target
+                field_exists = False
+                if "fields" in trigger_schema:
+                    field_exists = field_ref in trigger_schema["fields"]
+                    if (
+                        field_exists
+                        and "selector" in trigger_schema["fields"][field_ref]
+                    ):
+                        # Check if the selector type is allowed for this context key
+                        field_selector_config = trigger_schema["fields"][field_ref][
+                            CONF_SELECTOR
+                        ]
+                        field_selector_class = selector.selector(field_selector_config)
+                        if field_selector_class.selector_type not in allowed_keys.get(
+                            context_key, []
+                        ):
+                            raise vol.Invalid(
+                                f"The context '{context_key}' for '{field_name}' references '{field_ref}', but '{context_key}' does not allow selectors of type '{field_selector_class.selector_type}'. Allowed selector types: {', '.join(allowed_keys.get(context_key, []))}"
+                            )
+                if not field_exists and "target" in trigger_schema:
+                    # Target is a special field that always exists when defined
+                    field_exists = field_ref == "target"
+                    if field_exists and "target" not in allowed_keys.get(
+                        context_key, []
+                    ):
+                        raise vol.Invalid(
+                            f"The context '{context_key}' for '{field_name}' references 'target', but '{context_key}' does not allow 'target'. Allowed selector types: {', '.join(allowed_keys.get(context_key, []))}"
+                        )
+
+                if not field_exists:
+                    raise vol.Invalid(
+                        f"Context reference '{field_ref}' for key '{context_key}' does not exist "
+                        f"in trigger schema fields or target"
+                    )
+
+    return trigger_schema
+
+
 FIELD_SCHEMA = vol.Schema(
     {
         vol.Optional("example"): exists,
         vol.Optional("default"): exists,
         vol.Optional("required"): bool,
         vol.Optional(CONF_SELECTOR): selector.validate_selector,
+        vol.Optional("context"): {
+            str: str
```

Comment:
Please add a comment what the key and value in the context dict will be used for in the frontend.